state_machine/applications/system-check/tests/coil_test.py

```diff
@@ -0,0 +1,122 @@
+"""
+Python system check script for PyCubed satellite board
+PyCubed Mini mainboard-v02 for Pocketqube Mission
+Torque Driver Test
+* Author(s): Yashika Batra
+"""
+
+import time
+from lib import pycubed as cubesat
+from ulab.numpy.linalg import norm
+
+# voltage level constants; set between -1 and 1
+v1 = 0.25
+v2 = 0.5
+v3 = 0.75
+
+# find projected voltage: vlevel * max voltage
+projected_voltage1 = v1 * 5.06
+projected_voltage2 = v2 * 5.06
+projected_voltage3 = v3 * 5.06
+
+
+def user_test(coil_index):
+    """
+    All user interaction happens in this function
+    Set wait times, prompt user, collect magnetometer data for
+    different voltage levels
+    """
+
+    # wait times
+    wait_time = 3
+    driver_time = 3
+
+    # set up the test
+    print('Testing Coil Driver {coil_index} for the following voltage levels' +
+          'levels: {projected_voltage1}, {projected_voltage2}, ' +
+          '{projected_voltage3}.')
+    print("Waiting {wait_time} seconds.")
+    time.sleep(wait_time)
+
+    # conduct the test
+    print("Beginning the test now. Testing for {} seconds.".format(
+          driver_time * 3))
+
+    # test each voltage level
+    cubesat.coildriver_vout(coil_index, projected_voltage1)
+    time.sleep(driver_time)
+    mag_reading1 = cubesat.magnetic()
+
+    cubesat.coildriver_vout(coil_index, projected_voltage2)
+    time.sleep(driver_time)
+    mag_reading2 = cubesat.magnetic()
+
+    cubesat.coildriver_vout(coil_index, projected_voltage3)
+    time.sleep(driver_time)
+    mag_reading3 = cubesat.magnetic()
+
+    # calculate total magnetic reading
+    print("Data Collection Complete")
+    mag_total1 = norm(mag_reading1)
+    mag_total2 = norm(mag_reading2)
+    mag_total3 = norm(mag_reading3)
+
+    # if magnetometer readings are increasing over time, return true
+    result_val_bool = (mag_total3 > mag_total2 and mag_total2 > mag_total1)
+    if not result_val_bool:
+        print("Magnetometer results should have been increasing but were" +
+              "not. reading 1: {mag_total1}, reading 2: {mag_total2}," +
+              "reading 3: {mag_total3}")
+    return result_val_bool
+
+
+def voltage_levelx(result_dict, coil_index):
```

Comment:
Not super clear what this function is doing from its name.

Reply:
Agreed, I guess I made so many changes on that function that the name stopped being reflective of what it did. I've changed the name to coil_test